source/common/http/http1/codec_impl.cc

```diff
@@ -440,11 +440,11 @@ http_parser_settings ConnectionImpl::settings_{
     nullptr // on_chunk_complete
 };
 
-ConnectionImpl::ConnectionImpl(Network::Connection& connection, Stats::Scope& stats,
+ConnectionImpl::ConnectionImpl(Network::Connection& connection, CodecStats& stats,
                                http_parser_type type, uint32_t max_headers_kb,
                                const uint32_t max_headers_count,
                                HeaderKeyFormatterPtr&& header_key_formatter, bool enable_trailers)
-    : connection_(connection), stats_{ALL_HTTP1_CODEC_STATS(POOL_COUNTER_PREFIX(stats, "http1."))},
+    : connection_(connection), stats_(stats),
```

Comment:
I think you can change the definition of `stats_` to be a reference to avoid a copy of all elements. Same for h2?

Reply:
Yes, though I needed to make the FakeUpstream usage mirror the UpstreamImpl usage; basically all CodecStats are instantiated via the AtomicPtr now.